### PR TITLE
Update dev docs

### PR DESCRIPTION
**Pull request by victorlin**
### Description of proposed changes

Small touch ups. See commit messages.

### Related issue(s)

N/A

### Testing

N/A

### Checklist

- [x] ~Add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR that are end user focused. Keep headers and formatting consistent with the rest of the file.~ N/A, no functional changes

### CODE REVIEW

File: docs/contribute/DEV_DOCS.md [hunk 1]
```diff
@@ -1,18 +1,6 @@
...
-  - Running local code changes
-  - Testing
-  - Releasing
-  - Maintaining Bioconda package
-  - Continuous integration
-- Contributing documentation
-  - Formats (Markdown and reStructuredText)
-  - Documentation structure
-  - Building documentation
+Thank you for helping us to improve Augur! Use the GitHub markdown preview sidebar to navigate the sections in this document.
```
Comment (jameshadfield) [inline docs/contribute/DEV_DOCS.md:3]: I've always missed this from our docs. The original auspice docs had this, and when it works it's amazing (it only half works on GitHub).

**Review by jameshadfield — APPROVED** [review]

File: docs/contribute/DEV_DOCS.md (hunk 1 above)
Comment (victorlin) [reply docs/contribute/DEV_DOCS.md:3]: The GitHub preview sidebar works fine for me, but I'm not a huge fan of relying on it.

An embedded table of contents with links to each section (and only up to a certain depth) seems beneficial to me. There's VSCode extensions that automatically insert/maintain one, but that's relying on a tool that not everyone editing this file will use, and I'm not sure those can ignore sections after a certain depth.

I'd be open to manually adding/maintaining one at the top level, something like [this](https://github.com/nextstrain/augur/blob/f0449eca01a386f2e9c8cb12883dbac376e0ea07/docs/contribute/DEV_DOCS.md). Thoughts?

Comment (jameshadfield) [reply docs/contribute/DEV_DOCS.md:3]: Oh, I was referring to page designs which use the right margin to display a floating TOC, and then (bonus) highlight the entry you're currently reading. For example, [here's](https://developer.redis.com/develop/python/) a page built using docusaurus, which was what Auspice initially used for its docs.

Comment (victorlin) [reply docs/contribute/DEV_DOCS.md:3]: Oh! That does look much nicer. Probably not for this file unless we host it on RTD, but it sounds like some other of our RTD pages could benefit from this feature. Created an issue: https://github.com/nextstrain/docs.nextstrain.org/issues/161

**Pull request merged**